aiohttp/_http_parser.pyx

```diff
@@ -833,6 +834,8 @@ cdef parser_error_from_errno(cparser.llhttp_t* parser, data, pointer):
                  cparser.HPE_INVALID_TRANSFER_ENCODING}:
         return BadHttpMessage(err_msg)
     elif errno == cparser.HPE_INVALID_METHOD:
+        if b" " not in data:
+            return NotHttpProtocol(err_msg)
```

Comment:
This seems unreliable since a binary protocol can easily include `b" "` by accident.

A TLS handshake always begins with `b"\x16"` ([RFC 8446 §5.1](https://datatracker.ietf.org/doc/html/rfc8446#section-5.1)); we could use that to reliably detect HTTPS at least.

Reply:
Yeah its a bit messy to guess here... maybe we should just leave it as-is.

Reply:
I think its better to leave it as-is and not try to differentiate as llhttp is always going to be able to do a better job that a heuristic as otherwise we end up with a whole list of possible exceptions for TLS or other garbage